compat/src/suspense.js

```diff
@@ -128,15 +64,20 @@ Suspense.prototype = new Component();
  */
 Suspense.prototype._childDidSuspend = function(promise, suspendingInternal) {
 	const suspendingComponent = suspendingInternal._component;
+	if (suspendingComponent._onResolve != null) {
+		// This component has already been handled by a Suspense component. Do
+		// nothing
+		return;
+	}
```

Comment:
Since we re-render the suspending component component again when rendering the fallback, it'll throw again. This check ensures that additional throw doesn't duplicate the resolution logic. We already have logic to prevent re-entry (line 85: `if (resolved) return;`) but I like this additional check to prevent the number of promise callbacks from exploding.